src/main/java/gregtech/integration/jei/basic/GTFluidVeinCategory.java

```diff
@@ -0,0 +1,248 @@
+
+        // Vein Depletion Amount information
+        String veinDepletionAmount = I18n.format("gregtech.jei.fluid.depletion_amount") + " " + depletionAmount;
+        depletionAmountLength = minecraft.fontRenderer.getStringWidth(veinDepletionAmount);
+        minecraft.fontRenderer.drawString(veinDepletionAmount, textStartX, startPosY + 4 * FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Depleted Yield information
+        String veinDepletedYield = I18n.format("gregtech.jei.fluid.depleted_rate") + " " + depletedYield;
+        depletedYieldLength = minecraft.fontRenderer.getStringWidth(veinDepletedYield);
+        minecraft.fontRenderer.drawString(veinDepletedYield, textStartX, startPosY + 5 * FONT_HEIGHT + 1, 0x111111);
+
+        // Vein Dimensions information
+        String veinDimension = I18n.format("gregtech.jei.fluid.dimension") + " ";
+        int dimensionLength = minecraft.fontRenderer.getStringWidth(veinDimension);
+        minecraft.fontRenderer.drawString(veinDimension, textStartX, startPosY + 6 * FONT_HEIGHT + 1, 0x111111);
+
+        List<Integer> dimensionIDs = dimension.get();
+
+        //Will attempt to write dimension IDs in a single line, separated by commas. If the list is so long such that it
+        //would run off the end of the page, the list is continued on a new line.
+        int dimStartYPos = startPosY + 6 * FONT_HEIGHT + 1;
+        int dimDisplayPos = textStartX + dimensionLength;
+        for (int i = 0; i < dimensionIDs.size(); i++) {
+
+            //If the dimension name is included, append it to the dimension number
+            if (namedDimensions.containsKey(dimensionIDs.get(i))) {
+                dimName = namedDimensions.get(dimensionIDs.get(i));
+                fullDimName = i == dimensionIDs.size() - 1 ?
+                        dimensionIDs.get(i) + " (" + dimName + ")" :
+                        dimensionIDs.get(i) + " (" + dimName + "), ";
+            }
+            //If the dimension name is not included, just add the dimension number
+            else {
+
+                fullDimName = i == dimensionIDs.size() - 1 ?
+                        Integer.toString(dimensionIDs.get(i)) :
+                        dimensionIDs.get(i) + ", ";
+            }
+
+            //Find the length of the dimension name string
+            dimDisplayLength = minecraft.fontRenderer.getStringWidth(fullDimName);
+
+            //If the length of the string would go off the edge of screen, instead increment the y position
+            if (dimDisplayLength > (176 - dimDisplayPos)) {
+                dimStartYPos = dimStartYPos + FONT_HEIGHT;
+                dimDisplayPos = 70;
+            }
+
+            minecraft.fontRenderer.drawString(fullDimName, dimDisplayPos, dimStartYPos, 0x111111);
+
+            //Increment the dimension name display position
+            dimDisplayPos = dimDisplayPos + dimDisplayLength;
+        }
```

Comment:
This whole section is really similar to implementation in GTOreCategory and could be maybe extracted out and parametrized to work for both.